s3/aide.go

```diff
@@ -26,6 +28,15 @@ type Service struct {
 	svc s3iface.S3API
 }
 
+type ReaderService interface {
+	Read(string) (*io.ReadCloser, error)
+}
```

Comment:
This is an exported type and needs documentation. More importantly, I don't see why not just return the concrete type that implements the interface, rather than returning the interface.